# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/MediaEngineDefault.h
# Commit: 1248d2f659a9
# Full Hash: 1248d2f659a9f341f0de5d6558eecf364fc6f0a2
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2019-03-29 21:53:01
# Regressor Bug: 1494675
# File Overlap Count: 1
# Description:
#   Bug 1494675 - Remove windowID to MediaEngineSource mappings from MediaEngines. r=padenot
#   
#   These once served a caching-and-reuse purpose it seems, but it makes less sense
#   when sources are not shared.
#   
# ==============================================================================

diff -r 20543d03a4cd -r 1248d2f659a9 dom/media/webrtc/MediaEngineDefault.h
--- a/dom/media/webrtc/MediaEngineDefault.h	Wed Mar 27 14:10:24 2019 +0000
+++ b/dom/media/webrtc/MediaEngineDefault.h	Wed Mar 27 14:10:10 2019 +0000
@@ -127,8 +127,6 @@
       const nsTArray<const NormalizedConstraintSet*>& aConstraintSets,
       const nsString& aDeviceId) const override;
 
-  bool IsAvailable() const;
-
  protected:
   ~MediaEngineDefaultAudioSource();
 
@@ -147,18 +145,10 @@
 
   void EnumerateDevices(uint64_t aWindowId, dom::MediaSourceEnum, MediaSinkEnum,
                         nsTArray<RefPtr<MediaDevice>>*) override;
-  void Shutdown() override;
-  void ReleaseResourcesForWindow(uint64_t aWindowId) override;
+  void Shutdown() override {}
 
  private:
   ~MediaEngineDefault() = default;
-
-  // WindowID -> Array of devices.
-  nsClassHashtable<nsUint64HashKey, nsTArray<RefPtr<MediaEngineSource>>>
-      mVSources;
-  nsClassHashtable<nsUint64HashKey,
-                   nsTArray<RefPtr<MediaEngineDefaultAudioSource>>>
-      mASources;
 };
 
 }  // namespace mozilla